ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,203 @@
+      });
+    } catch (err) {
+      workerPool.config = value;
+
+      this.setState({
+        workerPool,
+        invalidProviderConfig: true,
+      });
+    }
+  };
+
+  render() {
+    const { isNewWorkerPool, classes } = this.props;
+    const { workerPool, invalidProviderConfig } = this.state;
+
+    return (
+      <Dashboard
+        title={
+          isNewWorkerPool
+            ? 'Worker Manager: Create Worker Type'
+            : 'Worker Manager: Edit Worker Type'
+        }>
+        <TextField
+          label="Enter Worker Type Name..."
+          name="name"
+          error={
+            Boolean(workerPool.name) && !isWorkerTypeNameValid(workerPool.name)
+          }
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.name}
+          margin="normal"
+        />
+
+        <TextField
+          label="Enter Worker Type Description..."
+          name="description"
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.description}
+          margin="normal"
+        />
+
+        <TextField
+          label="Enter Owner's Email..."
+          name="owner"
+          error={Boolean(workerPool.owner) && !workerPool.owner.includes('@')}
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.owner}
+          margin="normal"
+        />
+
+        <FormGroup>
+          <FormControlLabel
+            control={
+              <Switch
+                checked={workerPool.wantsEmail}
+                onChange={this.handleSwitchChange}
+                value="wantsEmail"
+              />
+            }
+            label="Receive emails about errors"
+          />
+        </FormGroup>
+
+        <FormGroup classes={classes.group}>
+          <FormLabel component="provider">Provider:</FormLabel>
+          <TextField
+            id="select-provider-type"
+            select
+            label="Type:"
+            helperText="Which service do you want to run your tasks in?"
+            value={workerPool.providerType}
+            name="providerType"
+            onChange={this.handleInputChange}
+            margin="normal">
+            <MenuItem value="gcp">GCP</MenuItem>
+          </TextField>
+
+          <TextField
+            label="Name:"
+            value={workerPool.providerId}
+            name="providerId"
+            onChange={this.handleInputChange}
+            margin="normal"
+          />
+
+          <List>
+            <ListItem>
+              <FormLabel component="config">Configuration:</FormLabel>
+            </ListItem>
+            <ListItem>
+              <CodeEditor
+                value={JSON.stringify(workerPool.config, null, 2)}
+                onChange={this.handleEditorChange}
+                lint
+              />
+            </ListItem>
+          </List>
+        </FormGroup>
+        <Button
+          spanProps={{ className: classes.createIconSpan }}
```

Comment:
I figured the button to open up the editor requires auth... Now I am realizing that somebody could enter `create` manually to the URL and get into the editor. It seems to me that the route should require auth, and not the buttons. I'd try that instead, what do you think?

Reply:
I think that's a really great idea but might be outside the scope of this PR. Feel free to create an issue for that. It would be good to add that extra layer of security.

Right now, resources that are protected will throw an error if the user is missing scopes. So if somebody tries to view a worker pool for example and that resource requires scopes to be viewed, an error panel will be displayed to the user saying that they are missing scopes.

Btw, the main reason behind disabling buttons that require auth is to simply avoid having the user make a request to taskcluster only to be shown an error about missing scopes.

Reply:
> I think that's a really great idea but might be outside the scope of this PR. Feel free to create an issue for that. It would be good to add that extra layer of security.

Sounds good to me. I'll `requiresAuth` for now